src/main/java/htsjdk/variant/vcf/VCFPathReader.java

```diff
@@ -0,0 +1,183 @@
+                codec,
+                requireIndex);
+    }
+
+
+    /**
+     * Parse a VCF file and convert to an IntervalList The name field of the IntervalList is taken from the ID field of the variant, if it exists. if not,
+     * creates a name of the format interval-n where n is a running number that increments only on un-named intervals
+     *
+     * @param path
+     * @return
+     */
+    public static IntervalList fromVcf(final Path path) {
+        return fromVcf(path, false);
+    }
+
+    public static IntervalList fromVcf(final Path path, final boolean includeFiltered) {
+        final VCFPathReader vcfReader = new VCFPathReader(path, false);
+        final IntervalList intervalList = fromVcf(vcfReader, includeFiltered);
+        vcfReader.close();
+        return intervalList;
+    }
+
+    /**
+     * Converts a vcf to an IntervalList. The name field of the IntervalList is taken from the ID field of the variant, if it exists. If not,
+     * creates a name of the format interval-n where n is a running number that increments only on un-named intervals
+     * Will use a "END" tag in the info field as the end of the interval (if exists).
+     *
+     * @param vcf the vcfReader to be used for the conversion
+     * @return an IntervalList constructed from input vcf
+     */
+
+    public static IntervalList fromVcf(final VCFPathReader vcf) {
+        return fromVcf(vcf, false);
+    }
+
+    public static IntervalList fromVcf(final VCFPathReader vcf, final boolean includeFiltered) {
+
+        //grab the dictionary from the VCF and use it in the IntervalList
+        final SAMSequenceDictionary dict = vcf.getFileHeader().getSequenceDictionary();
+        final SAMFileHeader samFileHeader = new SAMFileHeader();
+        samFileHeader.setSequenceDictionary(dict);
+        final IntervalList list = new IntervalList(samFileHeader);
+
+        int intervals = 0;
+        for (final VariantContext vc : vcf) {
+            if (includeFiltered || !vc.isFiltered()) {
+                String name = vc.getID();
+                final Integer intervalEnd = vc.getCommonInfo().getAttributeAsInt("END", vc.getEnd());
+                if (".".equals(name) || name == null)
+                    name = "interval-" + (++intervals);
+                list.add(new Interval(vc.getContig(), vc.getStart(), intervalEnd, false, name));
+            }
+        }
+
+        return list;
+    }
+
+    /**
+     * Returns the VCFHeader associated with this VCF/BCF file.
+     */
+    public VCFHeader getFileHeader() {
+        return (VCFHeader) reader.getHeader();
+    }
+
+    /**
+     * Returns an iterator over all records in this VCF/BCF file.
+     */
+    @Override
+    public CloseableIterator<VariantContext> iterator() {
+        try {
+            return reader.iterator();
+        } catch (final IOException ioe) {
+            throw new TribbleException("Could not create an iterator from a feature reader.", ioe);
```

Comment:
Why is this code using `TribbleException` if it's not part of `htsjdk.tribble`? If you want an unchecked IO Exception you can use `UncheckedIOException`.

Reply:
this is a copy of the VCFFileReader...I made very few changes....I think unless we want a serious rewrite of htsjdk (which is out of scope for this PR) I'd vote for keeping the current (slightly annoying) behavior.

Reply:
OK.